### PR TITLE
System io

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: playground/src/main/kotlin/kotlinx/io/gzip/GzipOutput.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    override fun flush(source: Buffer, length: Int) {
+       for (i in 0 until length) {
+           deflater.write(source[i].toInt())
+       }
+    }
+
+    override fun close() {
+        deflater.close()
+        original.writeArray(baos.toByteArray())
+        original.flush()
```
Comment (qwwdfsad) [inline playground/src/main/kotlin/kotlinx/io/gzip/GzipOutput.kt]: This flush should be called by `close` automatically, looks like an oversight ib `Output` API

File: playground/src/main/kotlin/kotlinx/io/bytes/ByteArrayOutput.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        require(initialSize > 0)
+    }
+    private var array: ByteArray = ByteArray(initialSize)
+    private var size = 0
+
+    public fun toArray(): ByteArray = array.copyOf(size)
+
+    override fun flush(source: Buffer, length: Int) {
+        ensureCapacity(length)
+        for (i in 0 until length) {
```
Comment (fvasco) [inline playground/src/main/kotlin/kotlinx/io/bytes/ByteArrayOutput.kt]: Please consider the `repeat` function

File: playground/src/main/kotlin/kotlinx/io/gzip/GzipInput.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    }
+    private val deflater = InflaterInputStream(inputStream)
+
+    override fun closeSource() {
+    }
+
+    override fun fill(buffer: Buffer): Int {
+        repeat(buffer.size) {
+            val byte: Byte = deflater.read().toByte()
+            if (byte == (-1).toByte()) return it
```
Comment (fvasco) [inline playground/src/main/kotlin/kotlinx/io/gzip/GzipInput.kt]: Please consider an explicit parameter (`pos`, `i`) in a multi-line lambda

File: playground/src/main/kotlin/kotlinx/io/gzip/GzipOutput.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import java.io.*
+import java.util.zip.*
+
+class GzipOutput(private val original: Output) : Output() {
+    // Do not judge me
+    private val baos = ByteArrayOutputStream()
+    private val deflater = DeflaterOutputStream(baos)
+
+    override fun flush(source: Buffer, length: Int) {
+       for (i in 0 until length) {
```
Comment (fvasco) [inline playground/src/main/kotlin/kotlinx/io/gzip/GzipOutput.kt]: Please consider the `repeat` function

File: playground/src/main/kotlin/kotlinx/io/bytes/ByteArrayOutput.kt (hunk 2 above)
Comment (qwwdfsad) [reply playground/src/main/kotlin/kotlinx/io/bytes/ByteArrayOutput.kt]: This module is named `playground` to emphasize its (much) lower requirements for code quality.
It's neither production-ready nor feature-complete. 
We have no interest in bikeshedding implementation details, naming or code-style of this module

File: playground/build.gradle [hunk 5]
```diff
@@ -0,0 +1,7 @@
+apply plugin: 'kotlin'
```
Comment (e5l) [inline playground/build.gradle]: Consider configure `kotlin-multiplatform` plugin

Comment (qwwdfsad) [reply playground/build.gradle]: What for?

File: core/commonMain/src/kotlinx/io/Console.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package kotlinx.io
+
+public expect object Console {
```
Comment (e5l) [inline core/commonMain/src/kotlinx/io/Console.kt]: KDoc

**Pull request merged**